### PR TITLE
Source property optimization

### PR DESCRIPTION
**Pull request by computerquip-streamlabs**
This should fix the Elgato `Configure Video` button along with a myriad of other optimizations.

### CODE REVIEW

File: app/components/shared/forms/Input.ts [hunk 1]
```diff
@@ -407,6 +407,15 @@ export function setPropertiesFormData(obsSource: obs.ISource, form: TFormData) {
     }
   });
 
+  /* Don't fetch properties unless we use it. */
+  if (buttons.length !== 0)
+    properties = obsSource.properties;
```
Comment (andycreeth) [inline app/components/shared/forms/Input.ts]: I noticed you omit the curly braces a lot for single line if-statements.  We loosely follow Airbnb's style guide which says that curly braces should only be omitted if the entire statement is a single line.

Here are some examples of what is acceptable and what isn't: https://github.com/airbnb/javascript#blocks

And here's a good explanation of why this style can potentially lead to bugs down the road: https://medium.com/@jonathanabrams/single-line-if-statements-2565c62ff492

I believe Prettier should be doing this automatically, but if it isn't, let me know.

File: app/components/shared/forms/Input.ts [hunk 2]
```diff
@@ -417,38 +426,59 @@ export function setPropertiesFormData(obsSource: obs.ISource, form: TFormData) {
     }
   });
 
-  obsSource.update(settings);
+  /* Don't update unless we need to. */
+  if (formInputs.length === 0)
+    return;
```
Comment (andycreeth) [inline app/components/shared/forms/Input.ts]: Same as above.

**Comment by computerquip-streamlabs** [comment]
I've made some changes and reorganized the commits a little bit to closer reflect each title.

**Comment by holiber** [comment]
I can't change the camera resolution. Is that ok?

image

**Pull request merged**